spec/lib/session-manager.spec.js

```diff
@@ -0,0 +1,102 @@
+"use strict";
+
+describe("sessionManager", () => {
+    var factory, promiseMock, randomMock, storageService, storageServiceFactoryMock;
+
+    beforeEach(() => {
+        var hashMock, OtDateMock;
+
+        hashMock = require("../mock/hash-mock")();
+        OtDateMock = require("../mock/ot-date-mock")();
+        promiseMock = require("../mock/promise-mock")();
+        randomMock = require("../mock/random-mock")();
+        storageServiceFactoryMock = require("../mock/storage-service-factory-mock")();
+        storageService = storageServiceFactoryMock.instance;
+
+        factory = () => {
+            var config;
+
+            config = {
+                session: {
+                    idHash: {
+                        idHash: "goes here"
+                    },
+                    idLength: 10,
+                    lifetime: {
+                        lifetime: "goes here"
+                    },
+                    storagePrefix: "anything"
+                }
+            };
+
+            return require("../../lib/session-manager")(config, hashMock, OtDateMock, promiseMock, randomMock, storageServiceFactoryMock);
+        };
+    });
+    it("exposes known methods", () => {
+        expect(factory()).toEqual({
+            createAsync: jasmine.any(Function),
+            validateAsync: jasmine.any(Function)
+        });
+    });
+    it("configured the storage service correctly", () => {
+        factory();
+        expect(storageServiceFactoryMock).toHaveBeenCalledWith({
+            idHash: "goes here"
+        }, {
+            lifetime: "goes here"
+        }, "anything");
+    });
+    describe(".createAsync()", () => {
```

Comment:
Do we need a fail case here? I assume of S3 or something else fails that's already handled and we'd get a 500 back or something like that.